packages/concerto-core/lib/decoratormanager.js

```diff
@@ -253,6 +253,252 @@ class DecoratorManager {
         return newModelManager;
     }
 
+    /**
+     * Extracts all the decorator commands from all the models in modelManager
+     * @param {ModelManager} modelManager the input model manager
+     * @param {object} options - decorator models options
+     * @param {boolean} options.removeDecoratorsFromModel - flag to strip out decorators from models
+     * @param {string} options.locale - locale for extracted vocabulary set
+     * @returns {Object} a new model manager with the decorations removed and a list of extracted decorator jsons
+     */
+    static extractDecorators(modelManager,options) {
+        options = {
+            removeDecoratorsFromModel: false,
+            locale:'en',
+            ...options
+        };
+        const ast = modelManager.getAst(true);
+        const decoratedAst = JSON.parse(JSON.stringify(ast));
+        let extractionDictionary={};
+        const processedModels = decoratedAst.models.map((model)=>{
+            if (model.decorators && model.decorators.length>0){
+                extractionDictionary=this.constructDCSDictionary(extractionDictionary,model.namespace, model.decorators,'','');
+                if (options.removeDecoratorsFromModel){
+                    model.decorators=null;
+                }
+            }
+            const processedDecl = model.declarations.map((decl)=>{
+                if (decl.decorators) {
+                    extractionDictionary=this.constructDCSDictionary(extractionDictionary,model.namespace,decl.decorators,decl.name,'');
+                }
+                if (options.removeDecoratorsFromModel){
+                    decl.decorators=null;
+                }
+                if (decl.$class.endsWith('.MapDeclaration')) {
+                    if (decl.key){
+                        if (decl.key.decorators){
+                            extractionDictionary=this.constructDCSDictionary(extractionDictionary,model.namespace,decl.key.decorators,decl.name,'','KEY');
+                            decl.key.decorators=null;
+                        }
+                    }
+                    if (decl.value){
+                        if (decl.value.decorators){
+                            extractionDictionary=this.constructDCSDictionary(extractionDictionary,model.namespace,decl.value.decorators,decl.name,'','VALUE');
+                            decl.value.decorators=null;
```

Comment:
`undefined` is better aligned to JS best practices. Does it make sense to delete the decorators property entirely?